level1c4pps/vgac2pps_lib.py

```diff
@@ -323,9 +325,77 @@
             "slope": 1.002,
             "offset": -0.69,
             "comment": "Based on collocation data for VZA < 3 and SZA 0-180",
+        },
+    },
+    "KNMI": {
+        "r06": {
+            "viirs_channel": "M05",
+            "slope": 0.9401,
+            "offset": 0.629,
+            "comment": "VZA<60, SZA<60, Delta(VZA) < 5, Delta(Scat-angle) < 5",
+        },
+        "r09": {
+            "viirs_channel": "M07",
+            "slope": 0.9345,
+            "offset": -1.304,
+            "comment": "VZA<60, SZA<60, Delta(VZA) < 5, Delta(Scat-angle) < 5",
+        },
+        "tb37_night": {
+            "viirs_channel": "M12",
+            "slope": 0.9934,
+            "offset": 1.659,
+            "min_sunzenith": 89,
+            "comment": " VZA<60, SZA>95, Delta(VZA) < 5",
+        },
+        "tb37_day": {
+            "viirs_channel": "M12",
+            "slope": 0.9572,
+            "offset": 9.572,
+            "max_sunzenith": 80,
+            "comment": "VZA<60, SZA<60, Delta(VZA) < 5, Delta(Scat-angle) < 5",
+        },
+        "tb37_twilight": {
+            "viirs_channel": "M12",
+            "slope": 0.9753,
+            "offset": 5.6155,
+            "min_sunzenith": 80,
+            "max_sunzenith": 89,
+            "comment": "The linear average of the SBAFs for t37_day and t37_night. 80<= SZA <89",
+        },
+        "tb11": {
+            "viirs_channel": "M15",
+            "slope": 0.9986,
+            "offset": 0.600,
+            "comment": "",
+        },
+        "tb12": {
+            "viirs_channel": "M16",
+            "slope": 0.9943,
+            "offset": 1.328,
+            "comment": "",
+        },
+    },
+    "NN_v1": {
+
+        "datadir": "/nobackup/smhid20/proj/safcm/data/SBAFS_NN/",
+        "comment": "NN based on AVHRR and VGAC matchups using all AVHRR heritage channels"
         }
     }
-}
+
+
+def convert_to_noaa19_NN(scene, sbaf_version):
+    """Applies AVHRR SBAF to VGAC channels using NN approach"""
+
+    if sbaf_version == "NN_v1":
+        sbaf_nn_dir = SBAF[sbaf_version]['datadir']
+    else:
+        logger.error(f"Unrecognized NN version, {sbaf_version}")
```

Comment:
I think you need logger.exception here, that will also raise an exception, otherwise you will end up with an uncontrolled exception later